nautobot/docs/development/release-checklist.md

```diff
@@ -125,14 +125,22 @@ Please see the [official Poetry documentation on `version`](https://python-poetr
 
 ### Update the Changelog
 
-Update the release notes for the new version and commit these changes to the `develop` branch.
+Create a release branch off of `develop` (`git checkout -b release-1.4.3 develop`)
+
+Generate release notes with `towncrier build --version 1.4.3`. This will update the release notes in `nautobot/docs/release-notes/version-1.4.md`, stage that file in git and `git rm` all of the fragments if you answer yes to the towncrier prompt `Is it okay if I remove those files? [Y/n]:`.
```

Comment:
```suggestion
Generate release notes with `towncrier build --version 1.4.3` and answer `yes` to the prompt `Is it okay if I remove those files? [Y/n]:`. This will update the release notes in `nautobot/docs/release-notes/version-1.4.md`, stage that file in git, and `git rm` all of the fragments that have now been incorporated into the release notes.
```